### PR TITLE
Feature/group controller

### LINKED ISSUE #58: Group Controller
`GET` Ansicht mit Übersicht aller Dateien einer Gruppe (auf die der Nutzer Lesezugriff hat)
(`material1/group/{group_id}` -> `material1/dir/{dir_id}`)

* Dieser Aufruf verlinkt immer auf das Root-Verzeichnis
* Root-Verzeichnis wird erstellt, wenn nicht vorhanden

`POST` Durchsuchen einer Gruppe nach einem Dateinamen, Tags oder Inhalt (`material1/group/{group_id}/search`)

### PR DESCRIPTION
**Pull request by Segelzwerg**
Adds a controller for the `/material1/group`

closes #58

### CODE REVIEW

File: src/main/java/mops/presentation/GroupController.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+     * @param model   spring view model
+     * @param groupId the id of the group to be searched
+     * @return the route to the template 'directory'
+     */
+    @PostMapping(path = "/{groupId}/search")
+    public String searchFilesInGroup(KeycloakAuthenticationToken token,
+                                     Model model,
+                                     @PathVariable("groupId") int groupId) {
+        final Account account = AccountUtil.getAccountFromToken(token);
+        final FileQuery query = (FileQuery) model.getAttribute("searchQuery");
```
Comment (iTitus) [inline src/main/java/mops/presentation/GroupController.java]: You can get this FileQuery from a parameter as well.

File: src/test/java/mops/presentation/GroupControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+     */
+    @MockBean
+    private FileService fileService;
+
+    /**
+     * Necessary bean.
+     */
+    @Autowired
+    private WebApplicationContext context;
+
```
Comment (iTitus) [inline src/test/java/mops/presentation/GroupControllerTest.java]: double newline, please remove

File: src/test/java/mops/presentation/GroupControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    @Test
+    public void getAllFilesOfDirectory() throws Exception {
+        SecurityContextUtil.setupSecurityContextMock("userName", "userEmail@mail.de", Set.of("studentin"));
+        mvc.perform(get("/material1/group/1"))
+                .andExpect(status().is2xxSuccessful())
+                .andExpect(view().name("files"));
+    }
+
+    /**
+     * Tests if all files the correct view is called upon searching in a group.
```
Comment (iTitus) [inline src/test/java/mops/presentation/GroupControllerTest.java]: Copy-Paste Error

File: src/test/java/mops/presentation/GroupControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    @MockBean
+    private FileService fileService;
+
+    /**
+     * Necessary bean.
+     */
+    @Autowired
+    private WebApplicationContext context;
+
+
```
Comment (Segelzwerg) [inline src/test/java/mops/presentation/GroupControllerTest.java]:
```suggestion

```

File: src/test/java/mops/presentation/GroupControllerTest.java (hunk 3 above)
Comment (Segelzwerg) [inline src/test/java/mops/presentation/GroupControllerTest.java]:
```suggestion
     * Tests if the correct view is called upon searching in a group.
```

File: src/test/java/mops/presentation/GroupControllerTest.java (hunk 4 above)
Comment (iTitus) [reply src/test/java/mops/presentation/GroupControllerTest.java]: yeah

File: src/test/java/mops/presentation/GroupControllerTest.java (hunk 3 above)
Comment (iTitus) [reply src/test/java/mops/presentation/GroupControllerTest.java]: yes

File: config/pmd/pmd.xml [hunk 5]
```diff
@@ -30,10 +30,16 @@
     <rule ref="category/java/performance.xml"/>
 
     <rule ref="category/java/security.xml"/>
-
+    Comment
```
Comment (iTitus) [inline config/pmd/pmd.xml]:
```suggestion

```

**Pull request merged**